File: algorithmExamples/stack.queue/괄호회전하기.js
function solution(s) {
  if (!s.length) return -1;
  let answer = 0;
  for (let j = 0; j < s.length; j++) {
    let result = 0;
    s = s.slice(1, s.length) + s[0];
    const stack = [];
    for (let i = 0; i < s.length; i++) {
      if (stack[stack.length - 1] === map(s[i])) stack.pop();
      else stack.push(s[i]);
      if (!stack.length) result++;
      if (i === s.length - 1 && stack.length) result = 0;
    }
    answer = answer < result ? result : answer;
  }
  return answer;
}

function map(str) {
  switch (str) {
    case ")":
      return "(";
    case "}":
      return "{";
    case "]":
      return "[";
    default:
      return false;
  }
}

File: JAVA/programmers/lv2/1217/_lv2/괄호회전하기.java
package category.ns.programmersTMP._lv2;
import java.util.*;

public class 괄호회전하기 {

    class Solution {
        public int solution(String s) {
            int answer = 0;
            Stack<Character> st;
            for(int rot = 0; rot < s.length(); rot++){
                st = new Stack<>();
                for(int i = rot; i < rot+s.length(); i++){
                    char c = s.charAt(i%s.length());
                    if(st.isEmpty()) st.push(c);
                    else {
                        char nc = st.peek();
                        if((nc == '[' && c == ']')||(nc == '{' && c == '}')||(nc == '(' && c == ')')){
                            st.pop();
                        } else {
                            st.push(c);
                        }
                    }
                }
                if(st.isEmpty()) answer++;
            }
            return answer;
        }
    }
}
